doc/CMakeLists.txt

```diff
@@ -83,7 +83,17 @@ function(add_docset name)
   set(DOCSET_HTML_DIR ${CMAKE_BINARY_DIR}/html/${name})
   set(DOCSET_MAKE_DIRS ${DOCSET_BUILD_DIR};${DOCSET_SRC_DIR})
   set(DOCSET_CLEAN_DIRS ${DOCSET_BUILD_DIR};${DOCSET_HTML_DIR})
-  set(DOCSET_ENV DOXYGEN_EXECUTABLE=${DOXYGEN_EXECUTABLE})
+
+  set(DOCSET_ENV
+	DOXYGEN_EXECUTABLE=${DOXYGEN_EXECUTABLE}
+	# The nrf_security nrfxlib Kconfig sources willl source nrf
+    # Kconfigs and therefore need to know where ZEPHYR_NRF_MODULE_DIR
+    # is. It's not clear how other Kconfig env vars are added and why
+    # this needed to be added as a special case here. There could be a
+    # better way to do this, but the author does not know how.
+	ZEPHYR_NRF_MODULE_DIR=${NRF_BASE}
+  )
+
```

Comment:
this looks like a hack, why is this needed? If nrf is sourcing nrfxlib, why is an nrfxlib component sourcing nrf?

Reply:
NCSDK-22101 is to follow up on the hack

Reply:
So the question is, where in Kconfig is this used so that it affects documentation?

Reply:
related PR: https://github.com/nrfconnect/sdk-nrf/pull/11475

Reply:
and why is this needed? Isn't this already sourced in nrf?

Reply:
It is not sourced in nrf.

It is a hack. We need to preserve the Kconfig order to maintain the same defaults values of MBEDTLS configurations.

We will follow this up later by getting rid of the default order dependencies, but right now this would take too long time and we want to get nrf_security moved as soon as possible so that people can continue to work on nrf_security outside of nrfxlib.